### ISSUE TITLE
[python] Import is attempted twice if module raises an ImportError

### ISSUE THREAD

**Issue by bentheiii** [swig/swig#1343]
Some python modules may need to raise an ImportError when something goes wrong with their initialization. The problem is that when this happens to a module outside of a package, the exact same module will attempt to be imported twice:

```python
try:
    return importlib.import_module(mname)  # first exception happens here
except ImportError:
    return importlib.import_module('_foo') # if mname == '_foo', then the same import will be attempted again
```
The module initialization can be cumbersome and expensive. Also in python3 it results in a strange occurrence where an error is its own cause:

```
Traceback (most recent call last):
  File "foo.py", line 14, in swig_import_helper
    return importlib.import_module(mname)
  File "C:\python_envs\x64\3.7\lib\importlib\__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 670, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 583, in module_from_spec
  File "<frozen importlib._bootstrap_external>", line 1043, in create_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
ImportError: could not load library boring.dll

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "foo.py", line 17, in <module>
    _foo = swig_import_helper()
  File "foo.py", line 16, in swig_import_helper
    return importlib.import_module('_foo')
  File "C:\python_envs\x64\3.7\lib\importlib\__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 670, in _load_unlocked
  File "<frozen importlib._bootstrap>", line 583, in module_from_spec
  File "<frozen importlib._bootstrap_external>", line 1043, in create_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
ImportError: could not load library boring.dll
```

The solution is to check whether `mname` is equal to the module name and if so, to not attempt the second import
```python
try:
    return importlib.import_module(mname)
except ImportError:
    if mname == '_foo':
        raise
    return importlib.import_module('_foo')
```
This solution is python2 compatible

**Comment by bentheiii** [comment]
None of the issues you linked addresses this one. And what's wrong with a simple solution? As it stands, importing an error-raising package-less module will cause the exact same code to run twice. This solution simply checks that the .py module doesn't try the same thing twice.

**Comment by wsfulton** [comment]
I can see how your simple solution fixes this for the one import problem which you've outlined in this issue. However, it this issue is related to the wider set of problems caused by the presence of the exception block. The proposed solution in #848 is also a simple fix and will solve the wider set of problems including the one here.

**Comment by wsfulton** [comment]
Fixed, see #848.
